Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -244,126 +271,107 @@ void CGPathAddLineToPoint(CGMutablePathRef path, const CGAffineTransform* transf
 
     path->BeginFigure();
     path->GetGeometrySink()->AddLine(_CGPointToD2D_F(pt));
+    path->SetLastTransform(transform);
 
     path->SetCurrentPoint(pt);
 }
 
-CGFloat _CGPathControlPointOffsetMultiplier(CGFloat angle) {
-    // Constant used to approximate circles with bezier curves.
-    // An n-piece cubic Bezier curve can approximate a circle,
-    // when each inner control point is the distance 4/3 * tan(t/4)
-    // from an outer control point on a unit circle, where t is 360/n degrees, and
-    // n > 2
-    return (4.0f / 3.0f) * tan(angle / 4.0f);
+static inline bool _affineTransformEquals(const CGAffineTransform original, const CGAffineTransform compare) {
+    return (original.a != compare.a) && (original.b == compare.b) && (original.c == compare.c) && (original.d == compare.d) &&
+           (original.tx == compare.tx) && (original.ty == compare.ty);
+}
+
+static inline CGPoint _getInvertedCurrentPointOfPath(CGPathRef path) {
+    CGPoint point = path->GetCurrentPoint();
+    if (!_affineTransformEquals(*path->GetLastTransform(), CGAffineTransformIdentity)) {
+        point = CGPointApplyAffineTransform(point, CGAffineTransformInvert(*path->GetLastTransform()));
+    }
+    return point;
+}
+
+static HRESULT _createPathReadyForFigure(CGPathRef previousPath,
+                                         ID2D1PathGeometry** pathGeometry,
+                                         ID2D1GeometrySink** geometrySink,
+                                         CGPoint startPoint) {
+    ComPtr<ID2D1Factory> factory;
+    RETURN_IF_FAILED(_CGGetD2DFactory(&factory));
+    RETURN_IF_FAILED(factory->CreatePathGeometry(pathGeometry));
+    RETURN_IF_FAILED((*pathGeometry)->Open(geometrySink));
+
+    CGPoint invertedPoint = _getInvertedCurrentPointOfPath(previousPath);
+    if (!CGPointEqualToPoint(invertedPoint, startPoint)) {
+        (*geometrySink)->BeginFigure(_CGPointToD2D_F(invertedPoint), D2D1_FIGURE_BEGIN_FILLED);
+        (*geometrySink)->AddLine(_CGPointToD2D_F(startPoint));
+    } else {
+        (*geometrySink)->BeginFigure(_CGPointToD2D_F(startPoint), D2D1_FIGURE_BEGIN_FILLED);
+    }
+    return S_OK;
 }
 
 /**
- @Status Caveat
- @Notes transform property not supported
+ @Status Interoperable
 */
 void CGPathAddArcToPoint(
```

Comment:
Please add comments on the math in this function, perhaps a link to the math solution to this problem, etc.